File: update_docs.py
import os
import re

routes_dir = 'src/routes'
docs_dir = 'docs'

def get_routes():
    route_map = {}
    for filename in os.listdir(routes_dir):
        if not filename.endswith('.js'): continue
        filepath = os.path.join(routes_dir, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            endpoints = []
            for line in content.split('\n'):
                # match router.get('/path' or router.post('/path'
                match = re.search(r'router\.(get|post|put|delete|patch)\(([\'"`])(.*?)\2', line)
                if match:
                    method = match.group(1).upper()
                    path = match.group(3)
                    endpoints.append((method, path))
            route_map[filename] = endpoints
    return route_map

def check_docs(route_map):
    for doc in os.listdir(docs_dir):
        if not doc.endswith('.md'): continue
        docpath = os.path.join(docs_dir, doc)
        with open(docpath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        print(f"--- {doc} ---")
        if 'consumer' in doc:
            for route in route_map.get('wallet.routes.js', []):
                if route[1] not in content and 'wallet' not in content:
                    print(f"Missing wallet: {route}")
            for route in route_map.get('wishlist.routes.js', []):
                if route[1] not in content and 'wishlist' not in content:
                    print(f"Missing wishlist: {route}")
            for route in route_map.get('ticket.routes.js', []):
                if route[1] not in content and 'ticket' not in content:
                    print(f"Missing ticket: {route}")
        
        elif 'restaurant' in doc:
            for route in route_map.get('owner.routes.js', []):
                if route[1].replace(':restaurantId', '<restaurantId>') not in content and 'settlement' in route[1]:
                    print(f"Missing settlement: {route}")
                if route[1].replace(':restaurantId', '<restaurantId>') not in content and 'table' in route[1]:
                    print(f"Missing table: {route}")

        elif 'delivery' in doc:
            for route in route_map.get('partner.routes.js', []):
                if route[1] not in content:
                    print(f"Missing delivery partner route: {route}")

if __name__ == '__main__':
    routes = get_routes()
    check_docs(routes)

File: update_docs.js
const fs = require('fs');
const path = require('path');

const routesDir = path.join(__dirname, 'src/routes');
const docsDir = path.join(__dirname, 'docs');

function getRoutes() {
  const routeMap = {};
  const files = fs.readdirSync(routesDir);
  files.forEach(file => {
    if (!file.endsWith('.js')) return;
    const content = fs.readFileSync(path.join(routesDir, file), 'utf8');
    const endpoints = [];
    const lines = content.split('\n');
    lines.forEach(line => {
      const match = line.match(/router\.(get|post|put|delete|patch)\((['"`])(.*?)\2/);
      if (match) {
        endpoints.push({ method: match[1].toUpperCase(), path: match[3] });
      }
    });
    routeMap[file] = endpoints;
  });
  return routeMap;
}

function checkDocs(routeMap) {
  const docs = fs.readdirSync(docsDir);
  docs.forEach(doc => {
    if (!doc.endsWith('.md')) return;
    const content = fs.readFileSync(path.join(docsDir, doc), 'utf8');
    console.log(`\n--- ${doc} ---`);
    
    if (doc.includes('consumer')) {
      (routeMap['wallet.routes.js'] || []).forEach(r => {
        if (!content.includes(r.path)) {
          console.log(`Missing consumer route: ${r.method} /api/wallet${r.path === '/' ? '' : r.path}`);
        }
      });
      (routeMap['wishlist.routes.js'] || []).forEach(r => {
        if (!content.includes(r.path)) {
          console.log(`Missing consumer route: ${r.method} /api/wishlist${r.path === '/' ? '' : r.path}`);
        }
      });
      (routeMap['ticket.routes.js'] || []).forEach(r => {
        if (!content.includes(r.path)) {
          console.log(`Missing consumer route: ${r.method} /api/tickets${r.path === '/' ? '' : r.path}`);
        }
      });
      (routeMap['membership.routes.js'] || []).forEach(r => {
        if (!content.includes(r.path)) {
          console.log(`Missing consumer route: ${r.method} /api/membership${r.path === '/' ? '' : r.path}`);
        }
      });
    } else if (doc.includes('restaurant')) {
      (routeMap['owner.routes.js'] || []).forEach(r => {
        const checkPath = r.path.replace(':restaurantId', '<restaurantId>');
        if (!content.includes(checkPath)) {
          console.log(`Missing owner route: ${r.method} ${r.path}`);
        }
      });
    } else if (doc.includes('delivery')) {
      (routeMap['partner.routes.js'] || []).forEach(r => {
        if (!content.includes(r.path.replace(':id', '<orderId>'))) {
          console.log(`Missing delivery partner route: ${r.method} ${r.path}`);
        }
      });
    } else if (doc.includes('admin')) {
      (routeMap['admin.routes.js'] || []).forEach(r => {
        if (!content.includes(r.path.replace(':id', '<userId>').replace(':id', '<restaurantId>'))) {
          console.log(`Missing admin route: ${r.method} ${r.path}`);
        }
      });
    }
  });
}

const routes = getRoutes();
checkDocs(routes);
